### ISSUE TITLE
A problem with adding a product to a category

### ISSUE THREAD

**Issue by mnowakCode** [Smile-SA/elasticsuite#2767]
After migration to Magento 2.4.4 and upgrading Elasticsuite to version 2.10.11, we saw a problem with adding products to categories. This problem shows only in the production environment and not always.
When we try to add any product to a category sometimes the rest products are gone, but the product is showing. After refreshing the page, the products are displayed again.

### Preconditions

Magento Version: Comerce, 2.4.4
ElasticSuite Version: 2.10.11
Environment: Production mode

### Steps to reproduce
1. Choose a category that has some products
2. Add a product to the category (try several times)

### Expected result
1. The product is showing in the category

### Actual result
Before adding a product:
<img width="960" alt="image" src="https://user-images.githubusercontent.com/78545108/198550339-45998d60-fb91-466b-a200-34c43ea1bd5c.png">

After adding a product (sku: 001-010-000051):
<img width="994" alt="image" src="https://user-images.githubusercontent.com/78545108/198550022-6e82fc20-d125-48f6-973d-65b2ce3be81b.png">

**Comment by vahonc** [comment]
Hello @mnowakCode,

How are products displayed on the frontend side after you have caught this bug, in other words, when you see only the last added product in the Admin panel?

BR,
Vadym

**Comment by mnowakCode** [comment]
Hello @vahonc,

When products are gone, I don't save this category, because think I lose the data. So I refresh the page and I try to one more time add the product in the right way.

Best, Marcin

**Comment by vahonc** [comment]
Have you checked if there are no any errors in the Console of your browser when you have this bug? Also please checked Magento logs Maybe it's a cache issue of your browser or maybe you have some cache extension or configuration on your prod env.

BR,
Vadym

**Comment by mnowakCode** [comment]
I checked everything. Logs in the web console, Magento logs, and I nothing found.

Best, Marcin

**Comment by mnowakCode** [comment]
hello @vahonc,
Today I got an idea, that can be cron and indexer fault. Because we have set all indexes on "Update be schedule". Cron does indexer every minute so maybe when categories are indexing, shows a fault, and data are gone.
What do you think about this idea?

Best, Marcin

**Comment by vahonc** [comment]
Hi @mnowakCode 

I'll try to look at this problem more deeply and try to reproduce on my test environment.

BR,
Vadym

**Comment by vahonc** [comment]
@mnowakCode,

For this moment I was not able to reproduce your issue. Are you using a store with multi-websites?

For me, it's strange that after you refresh the page and try to add products one more time and it fixes your issue. So yes, it could be linked to the Indexer issue. You need to check the indices settings. Have you tried to check the status of the indexer when you fetch this bug?

Also, provide an output of this command:
`$ curl http://localhost:9200/_cat/indices?v`

BR,
Vadym

**Comment by mnowakCode** [comment]
@vahonc 
Yes, I use a store with multi-websites. I changed the indexers responsible for category indexing and I think this solved this issue, but I'll let you know.

Best, Marcin

**Comment by romainruaud** [comment]
any update about this @mnowakCode ?

**Comment by mnowakCode** [comment]
@romainruaud, I'm sorry for the late response, but we had black weeks on our webshop, so I can't test my idea.
I'll let you know after my testing on the production environment.

**Comment by mnowakCode** [comment]
@romainruaud in this week we changed the indexers' settings, but it doesn't help with our problem. If you have any idea how to repair this, please let me know.

**Comment by mnowakCode** [comment]
@vahonc if you need help with the investigation, let me know.

Best, Marcin

**Comment by mnowakCode** [comment]
@vahonc, @romainruaud I still have the same problem. Do you have any information from other developers about it?

**Comment by vahonc** [comment]
Hello @mnowakCode,

Nope, we don't get any reports from our customers. It could be an issue linked to some specific configuration of your environment.

Have you any details (new scenario, screens, errors) that can help us understand where is problem?

BR,
Vadym

**Comment by mnowakCode** [comment]
Hello @vahonc,

I'll describe the problem from the start. 
I have the problem only in the production environment (Magento 2.4.4). My webshop has 5 stores.
When I add any product to a category sometimes (not always) the rest of the products from the categories are gone.
I don't get any information in the web console and Magento logs. 
I don't save, and refresh the page. After that products are back, but without the product that I want to add.
We can schedule the meeting when I can show you the error on live.

Best regards, 
Marcin

**Comment by vahonc** [comment]
@mnowakCode,

Maybe I will try to have a look at this issue again but can't guarantee that it will quickly. If you are interested in dedicated (paid) support through professional services, please send an email to elasticsuite@smile.fr.

BR,
Vadym

**Comment by RadekBucki** [comment]
@vahonc  @mnowakCode I also have this issue and I think that I discovered when it happens. It always happens when you open Add/Remove products modal and there is no selected products in the brackets instead of products in grid:

<img width="423" alt="Image" src="https://github.com/user-attachments/assets/1ddfb8d2-1597-42fc-985e-cc71c311fbc5" />

When it looks like this there is no problem:

<img width="423" alt="Image" src="https://github.com/user-attachments/assets/5d2937eb-a5ae-44b0-aabc-7007135930a3" />

I can add that it happens from time to time when you open this modal very quickly after page is loaded.

I found it while debugging in this moment selected item ids from that grid are readed `vendor/magento/module-visual-merchandiser/view/adminhtml/web/js/add_products.js`. It's probably some case of to slow data sync because originally in magento this grid was not in a popup:

<img width="1331" alt="Image" src="https://github.com/user-attachments/assets/335d3e28-ff2f-4edd-a1c0-ab1c0eb9eb24" />
